### PR TITLE
katawa-shoujo-re-engineered: init at 1.4.4

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    repo = "ksre";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-RYJM/wGVWqIRZzHLUtUZ5mKUrUftDVaOwS1f/EpW6Tk=";
+  };
+
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    echo "#!/${stdenvNoCC.shell}
```
Comment (nevivurn) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:29]: I wonder if there is any preference between this and `makeWrapper ... --add-flags ${src} --add-flags run`.

**Review by nevivurn — APPROVED** [review]
Result of `nixpkgs-review pr 297572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>katawa-shoujo-re-engineered</li>
  </ul>
</details>

Also works fine.

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix (hunk 1 above)
Comment (quantenzitrone) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:29]: Let us wonder together
I have no idea, but to me the makeWrapper version looks cleaner.

There is however something that lets me write the executable as a nix string and pass it as file.
However, other than a few language specific examples, I couldn't find any documentations on it and I didn't want to dig through the stdenv.mkDerivation code.

Comment (quantenzitrone) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix]: i changed it to use makeWrapper and also added an icon and desktop file

**Review by nevivurn — APPROVED** [review]
Result of `nixpkgs-review pr 297572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>katawa-shoujo-re-engineered</li>
  </ul>
</details>

Also tested working.

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  makeWrapper,
+  renpy,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "katawa-shoujo-re-engineered";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    /*
+      # alternative source:
```
Comment (AndersonTorres) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:16]: what is the official one?

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    makeWrapper ${lib.getExe' renpy "renpy"} $out/bin/${finalAttrs.meta.mainProgram} \
+      --add-flags ${finalAttrs.src} --add-flags run
+    install -D $src/web-icon.png $out/share/icons/hicolor/512x512/apps/${finalAttrs.meta.mainProgram}.png
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A fan-made modernization of the classic visual novel Katawa Shoujo";
+    homepage = "https://www.fhs.sh/projects"; # or https://codeberg.org/fhs/katawa-shoujo-re-engineered
```
Comment (AndersonTorres) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:57]:
```suggestion
    homepage = "https://www.fhs.sh/projects";
```

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      exec = finalAttrs.meta.mainProgram;
+    })
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+  dontBuild = true;
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:44]:
```suggestion
  ];

  dontBuild = true;

  installPhase = ''
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:16]: The codeberg one is the main.

image

Comment (AndersonTorres) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:16]: Not at the current time.

`srcs` is usually employed when the source spreads over multiple places (e.g. engine + assets), but not for specifying backup mirrors in this fashion...

P.S.: given the recent XZ Incident, use the official one.

Comment (AndersonTorres) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:16]: Not at the current time.

`srcs` is usually employed when the source spreads over multiple places (e.g. engine + assets), but not for specifying backup mirrors in this fashion...

P.S.: given the recent XZ Incident, use the official one.

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    mkdir -p $out/bin
+    makeWrapper ${lib.getExe' renpy "renpy"} $out/bin/${finalAttrs.meta.mainProgram} \
+      --add-flags ${finalAttrs.src} --add-flags run
+    install -D $src/web-icon.png $out/share/icons/hicolor/512x512/apps/${finalAttrs.meta.mainProgram}.png
+
+    runHook postInstall
+  '';
+
+  meta = {
```
Comment (drupol) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:52]: Missing `meta.mainPlatforms` and `meta.sourceProvenance`.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix (hunk 5 above)
Comment (quantenzitrone) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:52]: you mean `meta.platforms`?
and isn't `fromSource` the default `sourceProvenance`

Comment (quantenzitrone) [reply pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:52]: i added `meta.platforms`

File: pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    domain = "codeberg.org";
+    owner = "fhs";
+    repo = "katawa-shoujo-re-engineered";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-RYJM/wGVWqIRZzHLUtUZ5mKUrUftDVaOwS1f/EpW6Tk=";
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = finalAttrs.meta.mainProgram;
```
Comment (AndersonTorres) [inline pkgs/by-name/ka/katawa-shoujo-re-engineered/package.nix:25]: Write it directly instead.

```suggestion
      name = "katawa-shoujo-re-engineered";
```

**Review by drupol — APPROVED** [review]
LGTM, 2 minor things:

1. Move the creation of the desktop items before the meta attribute.
2. Use `makeBinaryWrapper` in place of `makeWrapper`

**Pull request merged**